core/src/main/java/com/linecorp/armeria/server/Routers.java

```diff
@@ -71,13 +76,44 @@
                             RejectedRouteHandler.class.getSimpleName(), e);
             }
         };
+
+        final Map<Route, ServiceConfig> newServiceConfigs = new HashMap<>();
+        final BiFunction<Route, ServiceConfig, ServiceConfig> fallbackValueConfigurator =
+                (originalRoute, fallbackServiceConfig) -> {
+                    final Route fallbackRoute = fallbackServiceConfig.route();
+
+                    if (originalRoute.complexity() == fallbackRoute.complexity() &&
+                        originalRoute.methods().containsAll(fallbackRoute.methods())) {
+                        return fallbackServiceConfig;
+                    }
+
+                    checkState(fallbackRoute.equals(Route.ofCatchAll()),
+                               "Fallback service must catch all requests.");
+                    final Route newRoute =
+                            originalRoute.toBuilder()
```

Comment:
@trustin I need to create a new fallback route from here. Is it okay to remove `final` keyword from `RouteBuilder#isFallback` variable and make a setter for it?

Reply:
SGTM